docs/respondent_management_to_electronic_questionnaire.rst

```diff
@@ -38,17 +38,33 @@ The following metadata keys are always required for the survey runner, they do n
 ``account_service_url``
   The url of the account service used to launch the survey
 
-**Schema Selection Fields**
+Schema Selection Fields
+^^^^^^^^^^^^^^^^^^^^^^^
+
+The schema selection fields determine the mechanism used by survey runner to load the questionnaire schema JSON.
 
 ``schema_name``
-  The eQ schema to launch
+ The name of the schema.
 ``form_type``
   The particular form_type for a responding unit
 ``eq_id``
   The eQ questionnaire id
+``schema_url``
+	A URL for a remote survey JSON. This claim is used to tell Survey Runner to load the schema JSON from a remote location
+
+The schema used by an eQ can be selected one of three ways.
+
+In priority order:
+
+#. ``schema_url`` with ``schema_name``
+	Launch eQ using a remote survey JSON. A ``schema_name`` is also required.
```

Comment:
When the `schema_url` is included, does the (then mandatory) `schema_name` have any functional significance or is just required as it need to be included in the submission data?

Reply:
It does currently. All in all, runner needs it for:
1. Logging
2. PDF file name
3. Downstream submission

Reply:
It might be a good time to revisit the PDF filename. I think there was probably zero thinking into the current naming.

For remote schemas the downstream submission should probably include the explicit URL of the schema and not just the `schema_name`. The `schema_name` just about works for EQ bundled schema but probably not for remote schema. One question I'd have is should a separate unique identifier be included _within_ authored schema, or is the `survey_url` enough? I certainly don't think a separate `schema_name` adds any value for remote schema.

There are other attributes available in a schema but I don't think [these values](https://github.com/ONSdigital/eq-questionnaire-schemas/blob/main/schemas/business/en/construction_0001.json#L6-L7) can be relied on for non-business survey schema.

From RAS's perspective, if the CI registry doesn't expose a `schema_name` like attribute per schema resource, what value to we use in the claims? It might have to be a hack of the `<file_name>.json`

Reply:
> It might be a good time to revisit the PDF filename. I think there was probably zero thinking into the current naming.

This is something I will be raising with Donna when she is back in, as this was never the ideal solution. As previously discussed, it should probably be the schema title + period for business surveys and something similar for adhoc.

> For remote schemas the downstream submission should probably include the explicit URL of the schema and not just the schema_name. The schema_name just about works for EQ bundled schema but probably not for remote schema. One question I'd have is should a separate unique identifier be included within authored schema, or is the survey_url enough? I certainly don't think a separate schema_name adds any value for remote schema.

Once `schema_name` reliance is removed from runner (PDF naming), going forward the plan from runner was to only need the name or the URL. Downstream would get one or the other based on the mechanism used. As for the identifier, that is something that could be passed to us, however, David Mort confirmed that the schema URL is immutable, but we need to confirm that no two schemas can have the same schema URL, if so it should be enough as an identifier. However, there could be an additional UUID identifier if needed. The identifier could also be part of the URL, hence avoid needing 2 claims.

> From RAS's perspective, if the CI registry doesn't expose a schema_name like attribute per schema resource, what value to we use in the claims? It might have to be a hack of the <file_name>.json

Right now, the `schema_name` is only needed when used with `schema_url` due to the PDF naming usage. Once we move away from that, it should be `schema_name` which matches a schema in the questionnaire schemas repo, or a URL.

Also, just to highlight, the purpose for this PR is just to document what currently happens. It will almost certainly change going forward, as changes are required as the CI work progresses. The idea was to document, what would happen now. My aim is for runner to only require a schema_name or a URL by the time the CI registry is ready for use.